### ISSUE TITLE
lerna produces JSONErrors that can't be logged by yargs

### ISSUE THREAD

**Issue by blahah** [lerna/lerna#811]
If there's a problem with the `package.json` of a package, `lerna ls` or `lerna bootstrap` throws an error that is about having an immutable error stack, rather than the actual JSONError.

## Expected Behavior

An invalid `package.json` for a package should cause `lerna` commands to throw a useful error, like this:

```bash
$ lerna ls
lerna info version 2.0.0-rc.4
lerna info versioning independent
/home/user/.config/yarn/global/node_modules/yargs/yargs.js:1078
      else throw err
           ^

TypeError: Property description must be an object: JSONError: Trailing comma in object at 14:3 in packages/ScienceReader/package.json
  },
  ^
    at module.exports (/home/user/.config/yarn/global/node_modules/parse-json/index.js:27:17)
    at parse (/home/user/.config/yarn/global/node_modules/load-json-file/index.js:8:29)
    at Function.module.exports.sync.fp [as sync] (/home/user/.config/yarn/global/node_modules/load-json-file/index.js:11:29)
    at Function.module.exports.sync (/home/user/.config/yarn/global/node_modules/lerna/node_modules/read-pkg/index.js:40:25)
    at /home/user/.config/yarn/global/node_modules/lerna/lib/PackageUtilities.js:109:47
    at Array.forEach (native)
    at /home/user/.config/yarn/global/node_modules/lerna/lib/PackageUtilities.js:103:86
...
```

## Current Behavior

This error is what is printed:

```bash
$ lerna bootstrap
lerna info version 2.0.0-rc.4
lerna info versioning independent
/home/user/.config/yarn/global/node_modules/yargs/yargs.js:1078
      else throw err
           ^

TypeError: Cannot set property stack of JSONError which has only a getter
    at EventEmitter.<anonymous> (/home/user/.config/yarn/global/node_modules/npmlog/log.js:185:17)
    at EventEmitter.<anonymous> (/home/user/.config/yarn/global/node_modules/npmlog/log.js:281:23)
    at BootstrapCommand.runPreparations (/home/user/.config/yarn/global/node_modules/lerna/lib/Command.js:241:26)
    at BootstrapCommand.run (/home/user/.config/yarn/global/node_modules/lerna/lib/Command.js:147:12)
    at Object.handler (/home/user/.config/yarn/global/node_modules/lerna/lib/commands/BootstrapCommand.js:57:45)
    at Object.self.runCommand (/home/user/.config/yarn/global/node_modules/yargs/lib/command.js:231:22)
    at Object.Yargs.self._parseArgs (/home/user/.config/yarn/global/node_modules/yargs/yargs.js:990:30)
    at Object.Yargs.self.parse (/home/user/.config/yarn/global/node_modules/yargs/yargs.js:533:23)
    at Object.<anonymous> (/home/user/.config/yarn/global/node_modules/lerna/bin/lerna.js:4:25)
    at Module._compile (module.js:571:32)
```

## Possible Solution

There are two possible solutions:

1. make lerna produce a JSONError that already has a mutable stack, so that yargs doesn't crash
2. Fix yargs so that it can assign stack to JSONErrors 

## Steps to Reproduce (for bugs)

1. Create a new lerna project
2. Create a package with an invalid `package.json`, e.g. with a trailing comma after the last entry
3. Run `lerna init && lerna bootstrap`

<details><summary>lerna.json</summary><p>

```json
<!-- Please paste your `lerna.json` here -->
{
  "lerna": "2.0.0-beta.38",
  "packages": [
    "packages/*"
  ],
  "version": "independent"
}
```
</p></details>

<details><summary>lerna-debug.log</summary><p>

</p></details>

## Your Environment

node v7.8

| OS | Version |
| --- | --- |
| Ubuntu | 16.04.2 |
-->

**Comment by blahah** [comment]
OK, I traced the source of the problem. The specific way `npm-log` [tries to stringify the error stack before logging](https://github.com/npm/npmlog/blob/master/log.js#L182-L185), combined with a change to the way Errors are made in node v7, causes the Error stack to sometimes to be immutable leading to the issue.

I initially though it was this issue [fixed in error-ex](https://github.com/Qix-/node-error-ex/issues/6#issuecomment-284100787) (which is used by read-pkg for package.json errors), but that fixed version is already included in lerna v1.0.0-rc4.

[Issue reported](https://github.com/npm/npmlog/issues/50) to `npm-log` with PR.

**Comment by evocateur** [comment]
Wow, thanks for digging into this! In the meant time, we could probably do much better with our "internal" error handling and reporting of those errors (_especially_ JSON-related errors, which I myself just suffered the other day while adding a new test fixture that I inadvertently left a trailing comma in...).
